### ISSUE TITLE
Delay connect until first channel is being listened to

### ISSUE THREAD

**Issue by denjaland** [laravel/echo#136]
Hi,

We're using Laravel Echo with pusher, and as you know, our subscription is concurrent-connection-limited.  Hence we're looking into delaying connection to Pusher until we have at least one channel to listen to.

Not all our vueJS components require listening to a channel, but we initialise the setup to pusher in our bootstrap.js file.  Hence, the connection is already opened, even though we're not necessarily listening to specific channels.

I have two routes: either I create a wrapper for laravel-echo, and add some logic there to keep track of how many channels we're actually listening to, and connect upon first channel listener being added / disconnect when last channel is left, or we'd be looking into creating a PR for laravel echo with some suggested changes.

I didn't find another case or issue for it yet, so apologies if it has been suggested before, but is this something others might be interested in?  Would this be a good addition to the package?

We're currently working on the wrapper, but might move our effort to creating a PR here instead.

**Comment by akizor** [comment]
I was in a similar situation. Had to fire initialization only when a user was authenticated and had the auth token available. Ended up creating my own service that was a wrapper for Echo and fired the initialization only on specific point. My service has to be maintained globally across components in order to share states of it. The same way I did with listening events. Fire them when needed, destroy them when not needed.
